### ISSUE TITLE
Blog UI improvements

### ISSUE THREAD

**Issue by charleslavon** [near/near-discovery#993]

**Comment by vanessadespain** [comment]
Hi Yall! I have some starting point feedback/questions. I'd love to provide more granular feedback once we align on some of these functional details and feasibility. 

# Blog MVP UI Feedback
[Figma Design here](https://www.figma.com/file/nqgzMvVASTSHVqUQPFQWsV/WORKING-%2F-Q1-'24-%2F-Website---v3.2024?type=design&node-id=585%3A29919&mode=design&t=R74NhW9NFGGPPlE9-1)

## Blog Landing Page

- Blog Posts become Cards, matching the design of the News cards from the home page. 
- How many posts appear on page load?
    - Can we start with 9 and lazy load using “Load More” button?
    - Can we have the latest post appear as a larger module at the top of the page? 
    - Stretch goal: This module could be manually controlled and admin can select a specific post to be featured here.

## Blog Post

- Are we locked into the layout inherited from the Posts or can we override this? 
    - Referencing: The author/date appearing above the image, the line at the left of the post
    ![image](https://github.com/near/near-discovery/assets/8346374/cee25ed5-82ef-4e76-bf48-00de51f2f288)
- [This](https://www.figma.com/file/nqgzMvVASTSHVqUQPFQWsV/WORKING-%2F-Q1-'24-%2F-Website---v3.2024?type=design&node-id=585%3A29426&mode=design&t=R74NhW9NFGGPPlE9-1) is the design I’d like us to get closer to
- I'm not able to see what the typography waterfall is currently (what H1, H2, H3, P is set to currently). We have these variables defined in the Design System (DIG). Is it okay if I provide feedback based on the Design System? 
    - TLDR: I'll provide a typography waterfall based on the DIG?
    - What are the Markdown variables that posts support/has stylistically defined? 
    - I’d like to start with
        - H1 [Title]
        - H2 
        - H3 
        - P [Body]
        - Link style
        - Caption style

**Comment by jackson-harris-iii** [comment]
Hey Vanessa, Thanks for the feedback! 

1. We can have the posts appear as cards, which elements in the figma are you referring to and I should be able to get what we have in line with the design

2. I think we would implement pagination like the existing blog, but infinite scroll could work here too as we already do that with existing posts.
  a. load more button is also an option, whichever makes the most sense. I'd say leveraging the infinite scroll is the easiest followed by the loads button and then pagination.

3. Providing feedback in that way is fine, but I am currently unfamiliar with it. (This can help bring me up to speed!)

4. We are not locked into the Post style layout. We're leveraging existing post widgets, but can create more blog centric ones to align with the link you provided.

5. As for the markdown here is a [link](https://www.markdownguide.org/cheat-sheet/) to what markdown supports. we do have some ability to style the supported elements as we see fit in our widget style which will be implemented with standard css.

**Comment by vanessadespain** [comment]
# Blog Post Design
I've updated the [Figma](https://www.figma.com/file/nqgzMvVASTSHVqUQPFQWsV/WORKING-%2F-Q1-'24-%2F-Website---v3.2024?type=design&node-id=585%3A29919&mode=design&t=YEw4xE54QeOXjpHZ-1) to show designs for each breakpoint and our typography

![image](https://github.com/near/near-discovery/assets/8346374/ebeb144d-3025-4dd1-a8f2-85c901c4a0ed)
1. Layout: 
   - On desktop the page text spans the 8 center columns of the page container. 
   - On smaller breakpoints the text spans the full page container width

![image](https://github.com/near/near-discovery/assets/8346374/f5ffceb0-e2e7-4e8b-9fe7-fbe9a7bb7bc5)
2. Cover image should be constrained to/cropped to a 2:1 aspect ratio. This will give us some visual consistency for the template.
3. Title text inherits the H1 style. All type styles are defined for desktop and mobile in [Figma](https://www.figma.com/file/nqgzMvVASTSHVqUQPFQWsV/WORKING-%2F-Q1-'24-%2F-Website---v3.2024?type=design&node-id=585%3A29919&mode=design&t=YEw4xE54QeOXjpHZ-1). 
4. Post details (author and date) should sit below the cover image. And remove the left-border-line inherited from the social post.

**Comment by charleslavon** [comment]
@vanessadespain @jackson-harris-iii  from those suggestion, here are the UI improvements I recommend we delay to post-MVP.   

1. Load more scrolls (for MVP we can limit the number of posts)
2. Latest post as the top featured (for MVP we can hard code a specific blog post, ie. chain abstraction)

Any objection or additional suggestions of things to save for later?

This way we can work toward deprecating wordpress faster, and finishing re-post.

**Comment by vanessadespain** [comment]
Sounds good to me!

**Comment by vanessadespain** [comment]
Hey @jackson-harris-iii , 

Reviewing the latest on [bosblog](https://near.org/bosblog). Here's some feedback to help guide improvements to match the designs. 

## Blog Landing Page: 
[Figma Link](https://www.figma.com/file/nqgzMvVASTSHVqUQPFQWsV/WORKING-%2F-Q1-'24-%2F-Website---v3.2024?type=design&node-id=585%3A29938&mode=design&t=wMRhE1UovKLYukwq-1)
![image](https://github.com/near/near-discovery/assets/8346374/c3930224-1fcc-4ad2-82d4-df578f41c507)

### Page layout grid: 
We should use the same page grid from marketing pages. On desktop the max-width of the page content is 1224px. Currently we're extending beyond this, a good measure is to look at the left and right edge of the navigation — page content should not extend beyond this. 
In production reference: 
![image](https://github.com/near/near-discovery/assets/8346374/cc28a34c-ea93-4e44-a121-c69a67faf89c)

### Latest Post: 
Should be 42px Medium FK Grotesk (in DIG "text-3XL-medium")

### Individual Post Cards: 
Instead of image and text side-by-side horizontally, these should stack vertically. [See original comment here](https://github.com/near/near-discovery/issues/993#issuecomment-1969385136)

**Post title** should be FK Grotesk, 24px Medium (in DIG "text-XL-medium")
**Post date** should be below post title, Mona Sans 14px Regular (in DIG "text-s-regular") using the date format Mon DD, YYYY (Aug 12, 2024) instead of 3/4/24. 

### Remove gray background
![image](https://github.com/near/near-discovery/assets/8346374/8c2c35a0-e9ec-4a55-8ae3-14ad8bbe16a6)

## Blog Post: 
[Figma Link](https://www.figma.com/file/nqgzMvVASTSHVqUQPFQWsV/WORKING-%2F-Q1-'24-%2F-Website---v3.2024?type=design&node-id=585%3A29426&mode=design&t=wMRhE1UovKLYukwq-1)
https://www.figma.com/file/nqgzMvVASTSHVqUQPFQWsV/WORKING-%2F-Q1-'24-%2F-Website---v3.2024?type=design&node-id=585%3A29426&mode=design&t=wMRhE1UovKLYukwq-1

### Page layout grid: 
We should use the same page grid from marketing pages. On desktop the max-width of the page content is 1224px. Currently we're extending beyond this, a good measure is to look at the left and right edge of the navigation — page content should not extend beyond this. 
In production reference: 
![image](https://github.com/near/near-discovery/assets/8346374/94f15433-6cdf-435d-8f85-50f2f879fecd)

• The image should fit the max-width of the page (1224px on desktop).

### Post Typography
[Link to the type waterfall](https://www.figma.com/file/nqgzMvVASTSHVqUQPFQWsV/WORKING-%2F-Q1-'24-%2F-Website---v3.2024?type=design&node-id=603%3A30842&mode=design&t=wMRhE1UovKLYukwq-1)
Title: Should be 42px FK Grotesk Medium (in DIG "text-3XL-medium")
Date: Should be 14px Mona Sans Regular (450) (in DIG "text-s-regular")
Post Author: Should be 14px Mona Sans Regular (450)  (in DIG "text-s-regular")
Post Author Handle: Should be 14px Mona Sans Bold (in DIG "text-s-bold")
Text link color: Should be hex6E56CF (in DIG "Violet-light-9")
Body text: Should be font-weight 450 Mona Sans 16px (in DIG "text-base")
List items: Add more spacing between line items when using the bulleted list <ol> <li>...</li></ol>. Can we add 8px margin-bottom to list items? Should help these be more legible. Can these use 14px Mona Sans Regular (450) (in DIG "text-s-regular")

### More Markdown Examples of Typography
I couldn't find post examples that use different headline styles (H1, H2, H3, etc) Is this something we can do so that we can demonstrate the possibilities of type sizes to the NF Marketing team? For example the [Etherland](https://near.org/bosblog?accountId=near&blockHeight=114529294) post could use H3s where we currently use bold text for headlines in the post body.

**Comment by jackson-harris-iii** [comment]
@vanessadespain The UI updates are live on prod now. Here's an example [post](https://near.org/near/widget/BlogPostPage?accountId=jacksonthedev.near&blockHeight=116080070&returnLocation=/near/widget/ProfilePage?accountId%3Dbasedjackson.near&tab=blog) can you give this another once over and let me know if there is any additional feedback you have. Thanks!

**Comment by vanessadespain** [comment]
Priority UI Update: 
@charleslavon  and I chatted about this one. He was able to show me the "Promote" flyout menu. We chatted about trying to simplify this a bit by combining the "Promote to blog" and "Repost to Activity Feed" into one menu. So the option would be "Repost" (or Promote if you decide to keep as is) and the secondary menu items would be "Activity Feed" and "My Blog". 
Current implementation: 
<img width="461" alt="image" src="https://github.com/near/near-discovery/assets/8346374/d559eef5-f3d3-4d98-8d3b-56ecb675a89c">

From Design: 
<img width="553" alt="image" src="https://github.com/near/near-discovery/assets/8346374/7d30651a-517d-4e3c-b141-c321348a6e0f">
In my design explorations I had looked at placing repost in the 3-dot menu and noticed the similarity between "Repost" and "Report" as well as decreased discoverability of placing Repost in this menu. I went for placing the Repost in the tray of icons alongside Comment, Share, etc. Food for thought if there's interest to push this a bit further. 


Nice to have UI Improvements to match Design: 
1. The width of the Title + Author section should match the blog post width. Instead the min-width seems to be determined by the length of the title. Longer titles DO match the width of the blog post. 
Short titles: 
![image](https://github.com/near/near-discovery/assets/8346374/f656f529-cef0-4b99-be2b-abb9182d6ecc)
![image](https://github.com/near/near-discovery/assets/8346374/5e14b20c-2aa9-4f9d-811b-05f9fde37b2d)
2. Date and Author text should be 14px (text-small)
3. Author name should be bold while the Author @ handle should be regular weight.
4. Body text should be font-weight 450
5. Smaller breakpoints should have an external margin of at least 24px so the Image and Back to Posts button don't touch the edge of the view port. In Design the smaller breakpoints the post body and the image have the same width. 
![image](https://github.com/near/near-discovery/assets/8346374/b171b665-5d16-43a5-95c2-187ee4da6d87)
6. I couldn't verify any other headline type styles in the page body.

**Comment by charleslavon** [comment]
Thanks for the guidance @vanessadespain.  @jackson-harris-iii let's go with the removal of the "Promote" menu item, replacing this with the "Repost" icon which has two options -> Activity Feed, My Blog
